From: =?utf-8?q?Lukas_M=C3=A4rdian?= <slyon@ubuntu.com>
Date: Thu, 19 May 2022 12:43:11 +0200
Subject: tests:ethernets: fix autopkgtest with alternating default value

---
 tests/integration/ethernets.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/integration/ethernets.py b/tests/integration/ethernets.py
index 06ac069..f2fcc4e 100644
--- a/tests/integration/ethernets.py
+++ b/tests/integration/ethernets.py
@@ -245,7 +245,8 @@ class _CommonTests():
         self.assertIn(b'tcp-segmentation-offload: on', out)
         self.assertIn(b'tx-tcp6-segmentation: on', out)
         self.assertIn(b'generic-segmentation-offload: on', out)
-        self.assertIn(b'generic-receive-offload: off', out)  # off by default
+        # enabled for armhf on autopkgtest.u.c but 'off' elsewhere
+        # self.assertIn(b'generic-receive-offload: off', out)
         # validate turning off
         with open(self.config, 'w') as f:
             f.write('''network:
